Framework/Algorithms/src/ConvertUnits.cpp

```diff
@@ -475,12 +465,16 @@ void ConvertUnits::convertViaTOF(Kernel::Unit_const_sptr fromUnit,
 
       /// @todo Don't yet consider hold-off (delta)
       const double delta = 0.0;
+      std::vector<double> vals(outputWS->x(i).begin(), outputWS->x(i).end());
       // Convert the input unit to time-of-flight
-      localFromUnit->toTOF(outputWS->dataX(i), emptyVec, l1, l2, twoTheta,
-                           emode, efixed, delta);
+      localFromUnit->toTOF(vals, emptyVec, l1, l2, twoTheta, emode, efixed,
+                           delta);
+
       // Convert from time-of-flight to the desired unit
-      localOutputUnit->fromTOF(outputWS->dataX(i), emptyVec, l1, l2, twoTheta,
-                               emode, efixed, delta);
+      localOutputUnit->fromTOF(vals, emptyVec, l1, l2, twoTheta, emode, efixed,
+                               delta);
```

Comment:
I think you should just refactor `toTOF` and `fromTOF` to accept `HistogramX` instead of `std::vector<double>`.